internal/controller/clusterorder_webhook.go

```diff
@@ -0,0 +1,45 @@
+package controller
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	cloudkitv1alpha1 "github.com/innabox/cloudkit-operator/api/v1alpha1"
+)
+
+func triggerWebHook(ctx context.Context, url string, instance *cloudkitv1alpha1.ClusterOrder) error {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Triggering webhook " + url)
+
+	jsonData, err := json.Marshal(instance)
+	if err != nil {
+		return fmt.Errorf("failed to marshal JSON: %w", err)
+	}
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Do(req)
```

Comment:
no need to retry if the call to the webhook fails because we will try to reconcile again later, is my understanding correct?

Reply:
I believe that returning a non-`nil` error from the Reconcile function triggers another run of the reconcile loop.